azure-secret-manager/src/main/java/io/micronaut/azure/secretmanager/AzureVaultConfigurationClient.java

```diff
@@ -0,0 +1,129 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.azure.secretmanager;
+
+import com.azure.security.keyvault.secrets.SecretClient;
+import io.micronaut.azure.secretmanager.client.SecretKeyvaultClient;
+import io.micronaut.azure.secretmanager.client.VersionedSecret;
+import io.micronaut.azure.secretmanager.configuration.AzureKeyvaultConfigurationProperties;
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.env.Environment;
+import io.micronaut.context.env.PropertySource;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.discovery.config.ConfigurationClient;
+import io.micronaut.scheduling.TaskExecutors;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Distributed configuration client implementation that fetches application secret values from Azure keyvalut.
+ */
+@Singleton
+@Requires(beans = SecretClient.class)
+@BootstrapContextCompatible
+public class AzureVaultConfigurationClient implements ConfigurationClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AzureVaultConfigurationClient.class);
+
+    private final AzureKeyvaultConfigurationProperties azureKeyvaultConfigurationProperties;
+    private final ExecutorService executorService;
+    private final SecretKeyvaultClient secretClient;
+
+    /**
+     * Default Constructor.
+     *
+     * @param azureKeyvaultConfigurationProperties Azure Secret Vault Client Configuration
+     * @param executorService                      Executor Service
+     * @param secretClient                         The secrets client
+     */
+    public AzureVaultConfigurationClient(
+            AzureKeyvaultConfigurationProperties azureKeyvaultConfigurationProperties,
+            @Named(TaskExecutors.IO) @Nullable ExecutorService executorService,
+            SecretKeyvaultClient secretClient) {
+        this.azureKeyvaultConfigurationProperties = azureKeyvaultConfigurationProperties;
+        this.executorService = executorService;
+        this.secretClient = secretClient;
+    }
+
+    @Override
+    public Publisher<PropertySource> getPropertySources(Environment environment) {
+
+        if (azureKeyvaultConfigurationProperties.getVaultURL() == null || azureKeyvaultConfigurationProperties.getVaultURL().equals("")) {
```

Comment:
you can use https://github.com/micronaut-projects/micronaut-core/blob/3.4.x/core/src/main/java/io/micronaut/core/util/StringUtils.java#L74